packages/assets-controllers/src/TokensController.test.ts

```diff
@@ -1365,6 +1387,62 @@ describe('TokensController', () => {
 
       generateRandomIdStub.mockRestore();
     });
+
+    it('stores multiple tokens from a batched watchAsset confirmation screen correctly when user confirms', async function () {
+      const generateRandomIdStub = jest
+        .spyOn(tokensController, '_generateRandomId')
+        .mockImplementationOnce(() => requestId)
+        .mockImplementationOnce(() => '67890');
+
+      const acceptedRequest = new Promise<void>((resolve) => {
+        tokensController.subscribe((state) => {
+          if (
+            state.allTokens?.[ChainId.mainnet]?.[interactingAddress].length ===
+            2
```

Comment:
```suggestion
            state.tokens.length === 2
```

maybe? but definitely keep the allTokens checks below